code/game/objects/items/devices/mobileticketplatform.dm

```diff
@@ -0,0 +1,239 @@
+/obj/item/device/mobileticketplatform
+	name = "handheld ticket dispenser"
+	desc = "A device that allows security personnel to issue tickes for violation of Space Law."
+	icon_state = "atmos"
+	item_state = "analyzer"
+	w_class = 2
+	slot_flags = SLOT_BELT
+
+	//Variables
+	var/ticketreason = null
+	var/ticketamount = null
+	var/screen = 0
+	var/ticketdescription = null
+	var/ticketid = null
+
+	var/obj/item/weapon/card/id/ID
+	var/datum/money_account/linked_account
+	var/datum/money_account/D
+	var/datum/spacelaw/C
+
+	var/list/comittedcrimes = list()
+	var/list/possible_violations = list()
+	var/printing = 0
+	req_access = list(access_brig)
+
+obj/item/device/mobileticketplatform/New()
+	..()
+	linked_account = department_accounts["Security"]
+
+/obj/item/device/mobileticketplatform/proc/reset() //Reset the device
+	ticketamount = null
+	ID = null
+	ticketreason = null
+	ticketdescription = null
+	ticketid = null
+	screen = 0
+	printing = 0
+
+/obj/item/device/mobileticketplatform/attackby(obj/O, mob/user, params) //Get credentials
+	if(istype(O, /obj/item/weapon/card/id))
+		visible_message("<span class='info'>[user] swipes a card through [src].</span>")
+		if(ID)
+			to_chat(user, "<span class='warning'>Idenfication Card already scanned!</span>")
+		else
+			ID = O
+			D = get_money_account(ID.associated_account_number)
+			to_chat(user, "<span class='warning'>Idenfication Card scanned!</span>")
+
+///////////////////////////////////////////////////////PAYMENT/////////////////////////////////////////////////
+
+/obj/item/device/mobileticketplatform/proc/pay_ticket(mob/user)
+	printing = 1
+	screen = 4
+	sleep(50)
+	if(linked_account)
+		if(ticketamount <= D.money)
+			playsound(src, 'sound/machines/chime.ogg', 50, 1)
+			visible_message("[bicon(src)] The [src] chimes.")
+			D.money -= ticketamount
+			linked_account.money += ticketamount
+
+			var/datum/transaction/T = new()
+			T.target_name = "[linked_account.owner_name]"
+			T.purpose = "[ticketreason]"
+			T.amount = "[ticketamount]"
+			T.source_terminal = src
+			T.date = current_date_string
+			T.time = worldtime2text()
+			D.transaction_log.Add(T)
+
+			T = new()
+			T.target_name = D.owner_name
+			T.purpose = "[ticketreason]"
+			T.amount = "[ticketamount]"
+			T.source_terminal = src
+			T.date = current_date_string
+			T.time = worldtime2text()
+			linked_account.transaction_log.Add(T)
+			print_ticket(usr)
```

Comment:
I can't remember entirely, but I was under the impression we should be using `user` over `usr`... you even are inconsistently using this over this PR.